api/uniter/application_test.go

```diff
@@ -20,89 +23,96 @@ import (
 type applicationSuite struct {
 	coretesting.BaseSuite
 
-	apiCaller basetesting.APICallerFunc
-	life      life.Value
-	statusSet bool
+	apiCallerFunc func(*gc.C) basetesting.APICallerFunc
+	life          life.Value
+	statusSet     bool
 }
 
 var _ = gc.Suite(&applicationSuite{})
 
 func (s *applicationSuite) SetUpTest(c *gc.C) {
 	s.BaseSuite.SetUpTest(c)
 	s.life = life.Alive
-	s.apiCaller = func(objType string, version int, id, request string, arg, result interface{}) error {
-		c.Assert(objType, gc.Equals, "Uniter")
-		switch request {
-		case "Life":
-			c.Assert(arg, gc.DeepEquals, params.Entities{Entities: []params.Entity{{Tag: "application-mysql"}}})
-			c.Assert(result, gc.FitsTypeOf, &params.LifeResults{})
-			*(result.(*params.LifeResults)) = params.LifeResults{
-				Results: []params.LifeResult{{
-					Life: s.life,
-				}},
-			}
-		case "Watch":
-			c.Assert(arg, gc.DeepEquals, params.Entities{Entities: []params.Entity{{Tag: "application-mysql"}}})
-			c.Assert(result, gc.FitsTypeOf, &params.NotifyWatchResults{})
-			*(result.(*params.NotifyWatchResults)) = params.NotifyWatchResults{
-				Results: []params.NotifyWatchResult{{
-					NotifyWatcherId: "1",
-				}},
-			}
-		case "CharmURL":
-			c.Assert(arg, gc.DeepEquals, params.Entities{Entities: []params.Entity{{Tag: "application-mysql"}}})
-			c.Assert(result, gc.FitsTypeOf, &params.StringBoolResults{})
-			*(result.(*params.StringBoolResults)) = params.StringBoolResults{
-				Results: []params.StringBoolResult{{
-					Result: "cs:mysql",
-					Ok:     true,
-				}},
-			}
-		case "CharmModifiedVersion":
-			c.Assert(arg, gc.DeepEquals, params.Entities{Entities: []params.Entity{{Tag: "application-mysql"}}})
-			c.Assert(result, gc.FitsTypeOf, &params.IntResults{})
-			*(result.(*params.IntResults)) = params.IntResults{
-				Results: []params.IntResult{{
-					Result: 1,
-				}},
+	s.apiCallerFunc = func(c *gc.C) basetesting.APICallerFunc {
```

Comment:
why isn't this just a method on the suite?

Reply:
yeah, it should be now that it's not an attribute